samples/bluetooth/direct_test_mode/src/fem/fem.h

```diff
@@ -0,0 +1,162 @@
+ *                         @ref NRF_RADIO_TASK_TXEN and Tx pin setting.
+ *
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int fem_tx_configure(uint32_t active_event, uint32_t deactive_event,
+		     uint32_t active_delay);
+
+/**@brief Set Front-end Module to RX mode (LNA).
+ *
+ * @note Transition from the Program State to the Receive State takes
+ *       specific settle time for device in microseconds and is triggered by
+ *       the LNA pin. Check your FEM specification to obtain that value.
+ *       Suitable timing can be achieved by the @p active_delay parameters.
+ *       This function triggers the NRF_RADIO_TASK_RXEN task when active_event
+ *       occurs or triggers it immediately when @ref FEM_EXECUTE_NOW is used.
+ *
+ * @param[in] active_event An event that triggers start of procedure - this
+ *                         event will be connected to appropriate PPI channel.
+ *                         @ref FEM_EXECUTE_NOW value causes start
+ *                         procedure immediately.
+ * @param[in] deactive_event An event that triggers deactivation of the Rx -
+ *                           this event will be connected to appropriate PPI
+ *                           channel.
+ * @param[in] active_delay Rx LNA activation delay between
+ *                         @ref NRF_RADIO_TASK_RXEN and Rx pin setting.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int fem_rx_configure(uint32_t active_event, uint32_t deactive_event,
+		     uint32_t active_delay);
+
+/**@brief Clears up the configuration provided by the @fem_tx_configure
+ *        or @ref fem_rx_configure.
+ */
+void fem_txrx_configuration_clear(void);
+
+/**@brief Stop Front-end Module RX/TX mode immediately.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int fem_txrx_stop(void);
+
+/**@brief Configures Tx gain of the Front-end Module in arbitrary units.
+ *
+ * Tx gain value might be set by a synchronous interface like for example
+ * the SPI interface in blocking mode.
+ *
+ * @param[in] gain Tx gain in arbitrary units.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int fem_tx_gain_set(uint32_t gain);
+
+/**@brief Calculate default Front-end module delay between PA/LNA GPIO pin activation and
+ *        radio readiness for reception or transmition.
+ *
+ * @param[in] rx Calculate a delay for radio reception. If set to false
+ *               calculate a delay for radio transmition.
+ * @param[in] mode Radio data rate and modulation.
+ *
+ * @return The activation delay value in microseconds.
+ */
+uint32_t fem_default_active_delay_calculate(bool rx,
+					    nrf_radio_mode_t mode);
+
+/**@brief Chooses one of two physical antenna outputs.
```

Comment:
Do all frontends have two physical antenna outputs? This function seems pretty specific for a generic FEM interface.

Reply:
Most of FEM has one or two physical antenna outputs. I don't know if there are any FEM for Bluetooth/802.15.4 with more antennas on the market. Anyway the devices that we support here don't have more than 2 physical antenna outputs